providers/src/airflow/providers/fab/auth_manager/fab_auth_manager.py

```diff
@@ -166,13 +169,36 @@ def get_cli_commands() -> list[CLICommand]:
             commands.append(GroupCommand(name="fab-db", help="Manage FAB", subcommands=DB_COMMANDS))
         return commands
 
+    def get_fastapi_app(self) -> FastAPI | None:
+        flask_blueprint = self.get_api_endpoints()
+
+        if not flask_blueprint:
+            return None
+
+        flask_app = create_app()
+        flask_app.register_blueprint(flask_blueprint)
+
+        app = FastAPI(
+            title="FAB auth manager API",
+            description=(
+                "This is FAB auth manager API. This API is only available if the auth manager used in "
+                "the Airflow environment is FAB auth manager. "
+                "This API provides endpoints to manager users and permissions managed by the FAB auth "
+                "manager."
+            ),
+        )
+        app.mount("/", WSGIMiddleware(flask_app))
+
+        return app
+
     def get_api_endpoints(self) -> None | Blueprint:
```

Comment:
I was thinking of making it private method whenever the Airflow 2 legacy UI is gone. We still need this one as long as we use the Airflow 2 legacy UI

Reply:
But if you prefer copying it in the method I can do that, I just feel having them separated in 2 methods makes more sense

Reply:
By "Airflow 2 legacy UI", do you mean when everything but the stuff for auth/plugins is gone, or completely gone?

Reply:
What is the difference between gone and completely gone? 😄 I actually was wrong, the method `get_api_endpoints` allow the auth manager to extend the Airflow 2 Rest API, so whenever the Airflow 2 Rest API is removed, it will be safe to remove this method as well from the interface (and then making private in the fab auth manager).

Reply:
With "gone", you dropped the "everything but the stuff for auth/plugins" before it :) I mean while we support the FAB auth manager or backcompat for plugins, vs when we can completely remove all traces of flask/fab.

I think this would be worth moving to fastapi now, since it's something we've added ourselves and not something we get from FAB directly.

Reply:
If we do that now, the Rest API related to users, roles, permissions etc wont work in main. I am fine doing it, we just might break some developer experience if doing so though.

Reply:
Fine for the short term.